lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -16,12 +16,13 @@ def run(&block)
           queue = build_queue
           lock = Mutex.new
 
-          @workers = thread_count.times.map do
-            Thread.new do
+          @workers = Array.new(thread_count) do
+            t = Thread.new do
               while (item = next_item(queue, lock))
                 yield item
               end
             end
+            t.abort_on_exception = true
```

Comment:
I guess this is more localized, but what's the downside of just putting `Thread.abort_on_exception = true` somewhere at startup?